connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -122,35 +160,56 @@ class UCSingleCatalog extends TableCatalog with SupportsNamespaces with Logging
 object UCSingleCatalog {
   val LOAD_DELTA_CATALOG = ThreadLocal.withInitial[Boolean](() => true)
   val DELTA_CATALOG_LOADED = ThreadLocal.withInitial[Boolean](() => false)
+
+  def generateCredentialProps(
+      scheme: String,
+      temporaryCredentials: TemporaryCredentials): Map[String, String] = {
+    if (scheme == "s3") {
+      val awsCredentials = temporaryCredentials.getAwsTempCredentials
+      Map(
+        // TODO: how to support s3:// properly?
```

Comment:
is this TODO still relevant? if so, please clarify what does "properly" mean? what is improper?

Reply:
This is kind of a hack that I don't fully understand. I have to configure `s3a` here while the user-specified path is `s3://`. Need some experts to take a look.

Reply:
The test only sets up fs implementation for `s3://`: https://github.com/unitycatalog/unitycatalog/blob/ceecee1977afb04ef7b3237071346782041607a3/connectors/spark/src/test/java/io/unitycatalog/spark/BaseSparkIntegrationTest.java#L50

Reply:
But can you write that down in the comment? that context needs to be written down for others to understand.

regarding the problem, why are you configuring s3a? is it because actual s3 support requires configuring s3a?

Reply:
I was just trying different approaches to make tests pass at that time, and using `s3a` happened to solve the issue. I can put my question in the comment `Why can't we use s3 instead of s3a?` but I really don't know the answer.

Reply:
I think it's pretty common to bring in `hadoop-aws` (which provides the s3a filesystem), and then just wire that up to `s3://` URIs in spark conf as well, i.e. `spark.hadoop.fs.s3.impl=org.apache.hadoop.fs.s3a.S3AFileSystem`.  In practice I don't think it's a problem to set the credential conf on both schemes just in case

Reply:
Cool! I can address this in a followup PR as this is unrelated to this PR.